velox/vector/tests/VectorRetainedSizeTest.cpp

```diff
@@ -0,0 +1,313 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+using namespace facebook;
+using namespace facebook::velox;
+
+namespace facebook::velox::test {
+namespace {
+
+class VectorRetainedSizeTest : public testing::Test,
+                               public velox::test::VectorTestBase {
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance(memory::MemoryManager::Options{});
+  }
+
+  uint64_t getToalStringBufferSize(const FlatVector<StringView>& vector) {
+    uint64_t size{0};
+    for (const auto& buf : vector.stringBuffers()) {
+      size += buf->capacity();
+    }
+    return size;
+  }
+
+  std::vector<std::string> longStrings_{
+      {std::string(100, 'a'), std::string(200, 'b'), std::string(300, 'c')}};
+};
+
+} // namespace
+
+TEST_F(VectorRetainedSizeTest, flatVectorNoStrings) {
+  auto vector = makeFlatVector<int32_t>(1'000, folly::identity);
+  uint64_t totalStringBufferSize = 0;
+  auto retainedSize = vector->retainedSize(totalStringBufferSize);
+
+  EXPECT_EQ(totalStringBufferSize, 0);
+  EXPECT_EQ(retainedSize, vector->retainedSize());
+}
+
+TEST_F(VectorRetainedSizeTest, flatVectorInlinedStrings) {
+  auto vector = makeFlatVector<std::string>(
+      1'000, [](auto row) { return std::string(row % 3, '.'); });
+  uint64_t totalStringBufferSize = 0;
+  auto retainedSize = vector->retainedSize(totalStringBufferSize);
+
+  EXPECT_EQ(totalStringBufferSize, 0);
+  EXPECT_EQ(retainedSize, vector->retainedSize());
+}
+
+TEST_F(VectorRetainedSizeTest, flatVectorMultipleStringBuffers) {
+  auto vector = makeFlatVector<StringView>({});
+  vector->resize(3);
+
+  std::vector<Buffer*> buffers;
+  buffers.resize(3);
+  buffers[0] = vector->getBufferWithSpace(longStrings_[0].size());
+  buffers[1] = vector->getBufferWithSpace(longStrings_[1].size());
+  buffers[2] = vector->getBufferWithSpace(longStrings_[2].size());
+
+  for (auto i = 0; i < 3; ++i) {
+    auto& str = longStrings_[i];
+    memcpy(buffers[i]->asMutable<char>(), str.data(), str.size());
+    vector->setNoCopy(0, StringView(buffers[i]->as<char>(), str.size()));
+  }
+
+  uint64_t totalStringBufferSize = 0;
+  auto retainedSize = vector->retainedSize(totalStringBufferSize);
+  uint64_t expectedBufferSize = getToalStringBufferSize(*vector);
+
+  EXPECT_EQ(totalStringBufferSize, expectedBufferSize);
+  EXPECT_EQ(retainedSize, vector->retainedSize());
+}
+
+TEST_F(VectorRetainedSizeTest, constantVectorWithStringBuffer) {
+  auto vector = makeConstant<StringView>(StringView(longStrings_[0]), 50);
```

Comment:
usage of longStrings_ makes it hard to read the tests; consider removing longStrings_